### PR TITLE
Slash commands for most fun commands

### PR DESCRIPTION
**Pull request by smartfrigde**
I moved most of the fun commands (everything except eco) to slash commands/interactions api.
xoxo

### CODE REVIEW

File: src/commands/fun/insult.ts [hunk 1]
```diff
@@ -1,5 +1,14 @@
...
... You are nothing to me but just another target. I will wipe you the fuck out with precision the likes of which has never been seen before on this Earth, mark my fucking words. You think you can get away with saying that shit to me over the Internet? Think again, fucker. As we speak I am contacting my secret network of spies across the USA and your IP is being traced right now so you better prepare for the storm, maggot. The storm that wipes out the pathetic little thing you call your life. You're fucking dead, kid. I can be anywhere, anytime, and I can kill you in over seven hundred ways, and that's just with my bare hands. Not only am I extensively trained in unarmed combat, but I have access to the entire arsenal of the United States Marine Corps and I will use it to its full extent to wipe your miserable ass off the face of the continent, you little shit. If only you could have known what unholy retribution your little "clever" comment was about to bring down upon you, maybe you would have held your fucking tongue. But you couldn't, you didn't, and now you're paying the price, you goddamn idiot. I will shit fury all over you and you will drown in it. You're fucking dead, kiddo.`
```
Comment (lexisother) [inline src/commands/fun/insult.ts:9]: This doesn't actually mention the user. Should it?

File: src/commands/fun/poll.ts [hunk 2]
```diff
@@ -1,7 +1,24 @@
...
 import {NamedCommand, RestCommand, poll, CHANNEL_TYPE, SendFunction, Command} from "onion-lasers";
+import {SlashCommandBuilder} from "@discordjs/builders";
+import {CommandInteraction} from "discord.js";
 import {pluralise} from "../../lib";
 
+export const header = new SlashCommandBuilder()
+    .setDescription("Create a poll.")
+    .addStringOption((option) => option.setName("question").setDescription("Question for the poll").setRequired(true))
+    .addIntegerOption((option) =>
+        option.setName("duration").setDescription("Duration of the poll in seconds.").setRequired(false)
```
Comment (lexisother) [inline src/commands/fun/poll.ts:11]: This description is wrong, as the bot actually takes the duration in *milliseconds.*
```js
const duration = options.getInteger("duration", false);
execSlashPoll(interaction, question, duration || 60000);
```
Please do a manual conversation of user input (e.g. `20` seconds) to milliseconds.

**Review by lexisother — CHANGES_REQUESTED** [review]
Most commands are functioning just fine, there's just a couple of things wrong.

File: src/commands/fun/poll.ts (hunk 2 above)
Comment (Hades785) [reply src/commands/fun/poll.ts:11]: Probably should put `parseDuration` elsewhere and use it for consistency.
https://github.com/keanuplayz/TravBot-v3/blob/e7217cecb5ee9793fd2fa4691de9a762911fd38b/src/commands/fun/modules/eco-bet.ts#L179-L196

Comment (lexisother) [reply src/commands/fun/poll.ts:11]: Good catch. @smartfrigde please move the `parseDuration` function into, say `src/lib.ts` and update all references to it accordingly. Use this function to parse the durations correctly.

File: src/commands/fun/insult.ts (hunk 1 above)
Comment (smartfrigde) [reply src/commands/fun/insult.ts]: Yes, fixed in latest commit.

File: src/commands/fun/poll.ts (hunk 2 above)
Comment (smartfrigde) [reply src/commands/fun/poll.ts:11]: Done in latest commit

**Review by lexisother — APPROVED** [review]
Looks good!

**Pull request merged**